api/src/config.rs

```diff
@@ -0,0 +1,1520 @@
+                return false;
+            }
+            if self.prefetch.threads == 0 || self.prefetch.threads > 1024 {
+                return false;
+            }
+        }
+
+        true
+    }
+}
+
+/// Configuration information for blob data prefetching.
+#[derive(Clone, Debug, Default, Deserialize, Eq, Hash, PartialEq, Serialize)]
+pub struct PrefetchConfigV2 {
+    /// Whether to enable blob data prefetching.
+    pub enable: bool,
+    /// Number of data prefetching working threads.
+    #[serde(default = "default_prefetch_threads")]
+    pub threads: usize,
+    /// The batch size to prefetch data from backend.
+    #[serde(default = "default_prefetch_batch_size")]
+    pub batch_size: usize,
+    /// Network bandwidth rate limit in unit of Bytes and Zero means no limit.
+    #[serde(default)]
+    pub bandwidth_limit: u32,
+    /// Prefetch all data from backend.
+    #[serde(default)]
+    pub prefetch_all: bool,
+}
+
+/// Configuration information for network proxy.
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub struct ProxyConfig {
+    /// Access remote storage backend via proxy, e.g. Dragonfly dfdaemon server URL.
+    #[serde(default)]
+    pub url: String,
+    /// Proxy health checking endpoint.
+    #[serde(default)]
+    pub ping_url: String,
+    /// Fallback to remote storage backend if proxy ping failed.
+    #[serde(default = "default_true")]
+    pub fallback: bool,
+    /// Interval for proxy health checking, in seconds.
+    #[serde(default = "default_check_interval")]
+    pub check_interval: u64,
+    /// Replace URL to http to request source registry with proxy, and allow fallback to https if the proxy is unhealthy.
+    #[serde(default)]
+    pub use_http: bool,
+}
+
+impl Default for ProxyConfig {
+    fn default() -> Self {
+        Self {
+            url: String::new(),
+            ping_url: String::new(),
+            fallback: true,
+            check_interval: 5,
+            use_http: false,
+        }
+    }
+}
+
+/// Configuration for registry mirror.
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub struct MirrorConfig {
+    /// Mirror server URL, for example http://127.0.0.1:65001.
+    pub host: String,
+    /// Ping URL to check mirror server health.
+    #[serde(default)]
+    pub ping_url: String,
+    /// HTTP request headers to be passed to mirror server.
+    #[serde(default)]
+    pub headers: HashMap<String, String>,
+    /// Whether the authorization process is through mirror, default to false.
+    /// true: authorization through mirror, e.g. Using normal registry as mirror.
+    /// false: authorization through original registry,
+    /// e.g. when using Dragonfly server as mirror, authorization through it may affect performance.
+    #[serde(default)]
+    pub auth_through: bool,
+    /// Interval for mirror health checking, in seconds.
+    #[serde(default = "default_check_interval")]
+    pub health_check_interval: u64,
```

Comment:
u64 seems too large for the check time interval?

Reply:
changed to u32